.github/workflows/docker.yml

```diff
@@ -0,0 +1,190 @@
+name: Build and Publish Docker Images
+
+on:
+  push:
+    paths:
+      - "**"
+      - "!docs/**"
+      - "!deploy/**"
+      - "!setup.py"
+      - "!.gitignore"
+      - "!.github/workflows/**"
+      - ".github/workflows/docker.yml"
+    branches:
+      - "*"
+    tags:
+      - "*.*.*"
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      image-name: ${{ steps.image.outputs.name }}
+      tags: ${{ steps.meta.outputs.tags }}
+      labels: ${{ steps.meta.outputs.labels }}
+      package-name: ${{ steps.package.outputs.name }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set lowercase image name
+        id: image
+        run: |
+          echo "name=${GITHUB_REPOSITORY,,}" >> $GITHUB_OUTPUT
+
+      - name: Set package name
+        id: package
+        run: |
+          echo "name=$(basename ${GITHUB_REPOSITORY,,})" >> $GITHUB_OUTPUT
+
+      - name: Extract metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ steps.image.outputs.name }}
+          tags: |
+            type=ref,event=branch,enable=${{ github.ref != 'refs/heads/main' && github.ref != 'refs/heads/master' }}
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=raw,value=nightly,enable={{is_default_branch}}
+
+      - name: Compute final tags
+        id: final-tags
+        run: |
+          readarray -t tags <<< "${{ steps.meta.outputs.tags }}"
+
+          if [[ "${{ github.ref_type }}" == "tag" ]]; then
+            tag="${{ github.ref_name }}"
+            if [[ "$tag" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
```

Comment:
Similar to the previous comment, do you think we should allow a leading `v`? Currently we don't do that, but I am hoping we switch to that in the future.

Reply:
it'd just be `v?` in front, which would trigger for "v1.0.0" and "1.0.0" but not for "myworld-1.0.0" nor "v1.0.0-rc1"